client/extensions/zoninator/state/locks/reducer.js

```diff
@@ -0,0 +1,67 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer, keyedReducer } from 'state/utils';
+import {
+	ZONINATOR_REQUEST_LOCK,
+	ZONINATOR_REQUEST_LOCK_ERROR,
+	ZONINATOR_UPDATE_LOCK,
+} from '../action-types';
+
+const isRequesting = createReducer(
+	{},
+	{
+		[ ZONINATOR_REQUEST_LOCK ]: () => true,
+		[ ZONINATOR_REQUEST_LOCK_ERROR ]: () => false,
+		[ ZONINATOR_UPDATE_LOCK ]: () => false,
+	}
+);
+
+export const requesting = keyedReducer( 'siteId', keyedReducer( 'zoneId', isRequesting ) );
+
+const isBlocked = createReducer(
+	{},
+	{
+		[ ZONINATOR_UPDATE_LOCK ]: () => false,
+		[ ZONINATOR_REQUEST_LOCK_ERROR ]: ( state, { blocked } ) => blocked,
+	}
+);
+
+export const blocked = keyedReducer( 'siteId', keyedReducer( 'zoneId', isBlocked ) );
+
+const lockInit = createReducer(
+	{},
+	{
+		[ ZONINATOR_UPDATE_LOCK ]: ( state, { reset } ) => ( reset ? new Date().getTime() : state ),
```

Comment:
On the issue of using timestamps in reducers, I've recently seen the following pattern #21114 to make the reducer pure (so it's completely determined by data it is fed by the action, rather than something opaque than `Date().getTime()` here). _tl;dr:_ Consider moving the `Date().getTime()` call to the corresponding action, and just pass the value to the reducer.